cpp/test/adapter-non-local.test.js

```diff
@@ -1,268 +1,671 @@
-//     .toString()
-
-//   const metadata = getMetadataSample()
-//   const operation = getOperationSample({
-//     amount: normalizedAmount,
-//     data: 'c0ffeec0ffeec0ffee',
-//     recipient: receiver.account,
-//   })
-
-//   const before = getAccountsBalances([receiver.account], [token, xerc20])
-
-//   // Fill in some tokens as collateral
-//   await token.contract.actions
-//     .transfer([user, lockbox.account, quantity, ''])
-//     .send(active(user))
-
-//   await adapter.contract.actions
-//     .settle([user, operation, metadata])
-//     .send(active(user))
-
-//   const after = getAccountsBalances([receiver.account], [token, xerc20])
-//   const receiverResults = receiver.contract.tables
-//     .results(getAccountCodeRaw(receiver.account))
-//     .getTableRow(0n)
-
-//   expect(
-//     substract(
-//       after[receiver.account][token.symbol],
-//       before[receiver.account][token.symbol],
-//     ).toString(),
-//   ).to.be.equal(quantity)
-
-//   expect(receiverResults).to.be.deep.equal({
-//     id: 0,
-//     data: operation.data,
-//   })
-// })
-//   })
-// })
+const { expect } = require('chai')
+const { Blockchain, expectToThrow, mintTokens } = require('@eosnetwork/vert')
+const { deploy } = require('./utils/deploy')
+const { Asset, Bytes, PublicKey } = require('@wharfkit/antelope')
+const R = require('ramda')
+const {
+  active,
+  precision,
+  getAccountCodeRaw,
+  getSymbolCodeRaw,
+  getSingletonInstance,
+  logExecutionTraces,
+  prettyTrace,
+} = require('./utils/eos-ext')
+const { getEventBytes, hexStringToBytes } = require('./utils/bytes-utils')
+const { substract, no0x } = require('./utils/wharfkit-ext')
+const { getAccountsBalances } = require('./utils/get-token-balance')
+const errors = require('./utils/errors')
+
+const ethers = require('ethers')
+const { evmOperationSamples, amounts, evmTopicZero, evmAdapter } = require('./samples/evm-operations')
+const { evmMetadataSamples, teeCompressedPubKey } = require('./samples/evm-metadata')
+const { trimPrecision } = require('./utils/precision-utils')
+const { symbolName } = require('typescript')
+
+const getSwapMemo = (sender, destinationChainId, recipient, data) =>
+  `${sender},${destinationChainId},${recipient},${R.isEmpty(data) ? '0' : '1'}`
+
+const attestation = 'deadbeef'
+const NULL_KEY = 'PUB_K1_11111111111111111111111111111111149Mr2R' // null initialization of public_key() CDT function
+
+describe('Adapter EVM -> EOS testing', () => {
+  const symbol = 'TST'
+  const evmTokenSymbol = 'TST'
+  const maxSupply = '500000000'
+  const localprecision = 6
+  const evmPrecision = 18
+  const fromEvmXerc20Precision = 8
+
+  const TABLE_STORAGE = 'storage'
+  const FEE_BASIS_POINTS = 1750
+  const FEE_BASIS_POINTS_DIVISOR = 1000000
+
+  const compressed = Uint8Array.from(
+    Buffer.from(
+      teeCompressedPubKey,
+      'hex',
+    ),
+  )
+  const teePubKey = PublicKey.from({ type: 'K1', compressed })
```

Comment:
This is not possible. It requires a Uint8Array and for the conversion I have to use Buffer, otherwise the data is not encoded as hex and the resulting pubKey would be a wrong one.

Reply:
Sorry I wasn't clear: I meant you can just factor out this code where you have saved `teeCompressedPubKey` and import it as a PublicKey from there